master/buildbot/reporters/bitbucket.py

```diff
@@ -124,26 +124,15 @@ def get_owner_and_repo(repourl):
         Takes a git repository URL from Bitbucket and tries to determine the owner and repository
         name
         :param repourl: Bitbucket git repo in the form of
-                    git@bitbucket.com:OWNER/REPONAME.git
-                    https://bitbucket.com/OWNER/REPONAME.git
-                    ssh://git@bitbucket.com/OWNER/REPONAME.git
+                    git@bitbucket.org:OWNER/REPONAME.git
+                    https://bitbucket.org/OWNER/REPONAME.git
+                    ssh://git@bitbucket.org/OWNER/REPONAME.git
+                    https://api.bitbucket.org/2.0/repositories/OWNER/REPONAME
         :return: owner, repo: The owner of the repository and the repository name
         """
-        parsed = urlparse(repourl)
+        m = re.search(r"^(?:git@|ssh://git@|https://(?:api\.)?)bitbucket\.\w+"
```

Comment:
I think let's keep this using urlparse. The regex hardcodes the domain and thus no longer supports custom base_url for self-hosted installations.